Userland/Libraries/LibTLS/TLSv12.cpp

```diff
@@ -468,38 +470,37 @@ Vector<Certificate> TLSv12::parse_pem_certificate(ReadonlyBytes certificate_pem_
 DefaultRootCACertificates::DefaultRootCACertificates()
 {
-    // FIXME: This might not be the best format, find a better way to represent CA certificates.
-    auto config_result = Core::ConfigFile::open_for_system("ca_certs");
-    if (config_result.is_error()) {
-        dbgln("Failed to load CA Certificates: {}", config_result.error());
+    auto cacert_result = Core::File::open("/etc/cacert.pem"sv, Core::File::OpenMode::Read);
+    if (cacert_result.is_error()) {
+        dbgln("Failed to load CA Certificates: {}", cacert_result.error());
         return;
     }
-    auto config = config_result.release_value();
-    reload_certificates(config);
+    auto cacert_file = cacert_result.release_value();
+    auto data_result = cacert_file->read_until_eof();
+    if (data_result.is_error()) {
+        dbgln("Failed to load CA Certificates: {}", data_result.error());
+        return;
+    }
+    auto data = data_result.release_value();
+    reload_certificates(data);
 }
 
-void DefaultRootCACertificates::reload_certificates(Core::ConfigFile& config)
+void DefaultRootCACertificates::reload_certificates(ByteBuffer& data)
 {
     m_ca_certificates.clear();
-    for (auto& entity : config.groups()) {
-        for (auto& subject : config.keys(entity)) {
-            auto certificate_base64 = config.read_entry(entity, subject);
-            auto certificate_data_result = decode_base64(certificate_base64);
-            if (certificate_data_result.is_error()) {
-                dbgln("Skipping CA Certificate {} {}: out of memory", entity, subject);
-                continue;
-            }
-            auto certificate_data = certificate_data_result.release_value();
-            auto certificate_result = Certificate::parse_asn1(certificate_data.bytes());
-            // If the certificate does not parse it is likely using elliptic curve keys/signatures, which are not
-            // supported right now. Currently, ca_certs.ini should only contain certificates with RSA keys/signatures.
-            if (!certificate_result.has_value()) {
-                dbgln("Skipping CA Certificate {} {}: unable to parse", entity, subject);
-                continue;
-            }
-            auto certificate = certificate_result.release_value();
-            m_ca_certificates.append(move(certificate));
+
+    auto certs = Crypto::decode_pems(data);
+
+    for (auto& cert_data : certs) {
+        auto certificate_result = Certificate::parse_asn1(cert_data.bytes());
+        // If the certificate does not parse it is likely using elliptic curve keys/signatures, which are not
+        // supported right now. It might make sense to cleanup cacert.pem before adding it to the system.
+        if (!certificate_result.has_value()) {
+            dbgln("Skipping CA Certificate: unable to parse");
+            continue;
         }
+        auto certificate = certificate_result.release_value();
+        m_ca_certificates.append(move(certificate));
```

Comment:
i feel you should validate here, at least that certificate has `ca` field set in it.

could you put a `TODO` or `FIXME` for that? (in fact thereʼs more to check, feel free to more details in the comment 😉)

Reply:
What do you mean by "has ca field" set? A lot of these will be self signed because they are root CAs.

Reply:
well… `certificate` is appended as one of `m_ca_certificate`. to be a ca certificate, a certificate needs to have a ca flag set in the basic constraints extension. besides it should also should match subject/issuer dn and verify itself with the public key being certified.

(not really sure if thatʼs something to be tackled in this pull request—feel free to do that, though!—thatʼs only a hint for `TODO`/`FIXME`.)

Reply:
I did add a check that it is a really a CA, I dont agree with the rest you said. That is imho nothings that needs to be checked here.

Reply:
well… no need to agree, of course. what iʼve read is that you work on “default _root CA certificates_”, where [root certificate](https://en.wikipedia.org/wiki/Root_certificate) is a well defined notion.

that must be just a misnomer here—i guess it should be more of trust store then. 🤔

assuming that iʼd even say: okay. 🙂

Reply:
I would say it is more of a trust store as users can put into it what they want.

The file we now use by default is extracted from the Firefox browser (exluding name contraints).

Reply:
shall i try to bait you into a name change or a `TODO` here or there? 😅

(resolve at will!)

Reply:
I added a fixme for things that we might want to check in the (near) future